diff -ur netgen-4.9.13/libsrc/general/ngexception.hpp netgen/libsrc/general/ngexception.hpp
--- netgen-4.9.13/libsrc/general/ngexception.hpp	Mon Aug 10 13:40:52 2009
+++ netgen/libsrc/general/ngexception.hpp	Wed Apr 13 10:57:39 2011
@@ -7,11 +7,22 @@
 /* Date:   16. Jan. 2002                                                  */
 /**************************************************************************/
 
+// Modifications for creating a DLL in Windows
+#ifdef DLL_EXPORT
+   #if defined(NGLIB_EXPORTS) || defined(nglib_EXPORTS)
+      #define DLL_HEADER   __declspec(dllexport)
+   #else
+      #define DLL_HEADER   __declspec(dllimport)
+   #endif
+#else
+   #define DLL_HEADER 
+#endif
+
 namespace netgen
 {
 
 /// Base class for all ng exceptions
-class NgException 
+class DLL_HEADER NgException 
 {
   /// verbal description of exception
   string what;
diff -ur netgen-4.9.13/libsrc/include/mydefs.hpp netgen/libsrc/include/mydefs.hpp
--- netgen-4.9.13/libsrc/include/mydefs.hpp	Thu May 20 23:07:04 2010
+++ netgen/libsrc/include/mydefs.hpp	Fri Apr 15 11:38:43 2011
@@ -19,16 +19,16 @@
 // in Linux, "PACKAGE_VERSION" is replaced 
 // in the configure/make phases, with the 
 // right version number
-#ifdef WIN32
+#if defined(WIN32) || !defined(PACKAGE_VERSION)
 #define PACKAGE_VERSION "4.9.13"
 #endif
 
 
-#ifdef WIN32
-    #define DLL_HEADER   __declspec(dllexport)
-#else
-    #define DLL_HEADER 
-#endif
+//#ifdef WIN32
+//    #define DLL_HEADER   __declspec(dllexport)
+//#else
+//    #define DLL_HEADER 
+//#endif
 
 
 
diff -ur netgen-4.9.13/libsrc/interface/writeuser.hpp netgen/libsrc/interface/writeuser.hpp
--- netgen-4.9.13/libsrc/interface/writeuser.hpp	Tue Oct 27 23:11:52 2009
+++ netgen/libsrc/interface/writeuser.hpp	Wed Apr 13 12:02:46 2011
@@ -58,6 +58,7 @@
 
 // Philippose - 29/01/2009
 // Added GMSH v2.xx Mesh Export support
+extern
 void WriteGmsh2Format (const Mesh & mesh,
                        const CSGeometry & geom,
                        const string & filename);
diff -ur netgen-4.9.13/nglib/nglib.h netgen/nglib/nglib.h
--- netgen-4.9.13/nglib/nglib.h	Tue May 18 13:20:26 2010
+++ netgen/nglib/nglib.h	Wed Apr  6 14:57:06 2011
@@ -23,8 +23,8 @@
 
 // Philippose - 14.02.2009
 // Modifications for creating a DLL in Windows
-#ifdef WIN32
-   #ifdef NGLIB_EXPORTS || nglib_EXPORTS
+#ifdef DLL_EXPORT
+   #if defined(NGLIB_EXPORTS) || defined(nglib_EXPORTS)
       #define DLL_HEADER   __declspec(dllexport)
    #else
       #define DLL_HEADER   __declspec(dllimport)
